### DISCUSSION TITLE
Reuse the v1 access token obtained via "3 legged" auth upon application restart

### DISCUSSION THREAD

**Discussion by tat2bu** [plhery/node-twitter-api-v2#311] [Q&A]
Hello,

Is there a way to reinitialize a client from scratch using the "3 legged authentication token" obtained [this way](https://github.com/PLhery/node-twitter-api-v2/blob/master/doc/auth.md#user-wide-authentication-flow)?

It feels it's doable using the 2.0 api refresh token, but that it's not possible to reinitialize a client from the retrieved values using the 1.0a auth flow.

Is there a way to handle gracefully service or application restart without asking a new PIN for the user upon application restart using 1.0a auth flow?

Many thanks for your insights!

**Comment by alkihis** [comment]
Hi,
At the end of the process you obtain two token, persistent access token and access secret.
Just create a new client like you do on step 2, but with the two persistent tokens instead of the temporary ones :)
